lib/reply.js

```diff
@@ -396,6 +438,20 @@ function onSendEnd (reply, payload) {
   const req = reply.request
   const statusCode = res.statusCode
 
+  // we check if we need to update the trailers header and set it
+  const trailerHeaders = Object.keys(reply[kReplyTrailers])
```

Comment:
Could you avoid Object.keys() if there are no trailers? Just check if it's not `null`.